eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java

```diff
@@ -2145,6 +2150,26 @@ public boolean isEmpty()
             return UnifiedMap.this.isEmpty();
         }
 
+        @Override
+        public K getFirst()
+        {
+            if (UnifiedMap.this.size() > 0)
+            {
+                return (K) this.toArray()[0];
+            }
+            return null;
+        }
+
+        @Override
+        public K getLast()
+        {
+            if (UnifiedMap.this.size() > 0)
+            {
+                return IterableIterate.getLast(this);
```

Comment:
@johnjstuart this is a bit dangerous in terms of performance because this iterates the whole collection, if we can we should find a better alternative I think